### DISCUSSION TITLE
Mono- or bilingual translation files?

### DISCUSSION THREAD

**Discussion by keunes** [WeblateOrg/weblate#6862] [Q&A] [answered]
Using [`mdpo`](https://github.com/mondeja/mdpo) I created a set of PO-files from a set of MD-files used to build our website:
https://github.com/keunes/AP-website-copy/tree/translation/_i18n/PO-files

Comparing these PO files to the [examples in the Weblate documentation](https://docs.weblate.org/en/latest/formats.html#gnu-gettext) I concluded that I have "bilingual gettext PO files".

However, when creating a resource with bilingual PO files in Weblate I get the following alert:
> ** Duplicated translation.**
> The component contains translation file for the source language.
> Please consider the following:
> * You might be using a monolingual translation file as a bilingual one. If that is the case, please set the monolingual base language file location.
> * In case translation to the source language is desired, please change the source language in the component settings.
> * If the translation file for the source language is not needed, please remove it from the repository.
> * If the translation file for the source language is needed, but should be ignored by Weblate, please adjust the language filter to exclude it.
> The following occurrences were found:

Language | Language codes
-- | --
English | en, en

However, when following the first suggestion (setting the base language file location) I get the error "You can not use a base file for bilingual translation."
![afbeelding](https://user-images.githubusercontent.com/11229646/143139496-aa938df2-ef9c-4b8c-a45b-d30b810d0645.png)

Should I change the File format to 'gettext PO file (monolingual)' even though my PO file is (does not seem to be) monolingual?

**Comment by nijel** [comment]
The file is bilingual (it contains source and target languages in a single file).

Why do you have `en` files when translating from English? If you need them for some reason, exclude them from Weblate via language filter. If you do not need them, remove them.

**Reply by keunes** [reply]
> Why do you have `en` files when translating from English?

That is a very good question I think :) We have a large set of MD files that are used to build our Jekyll website. We turn these into PO files using the tool [`mdpo`](https://github.com/mondeja/mdpo), for two reasons: 1) Weblate doesn't seem to have full support for MD files (yet) and 2) we don't want to have a resource per website page but rather one resource per website section (and in Weblate there can be only one file per resource).

So; that tool we use creates PO files from a bunch of MD files, and this bilingual file is what the tool outputs.

(Once translated through Weblate, the translated PO files are used to create the translated MD files.)

Not sure the tool `mdpo` can currently outup anything else. Would it work if I save the output as POT file instead?

**Reply by keunes** [reply]
Reading again your reply, I don't need them. But actually in Weblate the strings are empty because they are in the file - so it's not possible to e.g. 'copy from source'. Would there be a way to have the string ID taken as the string text, in Weblate?

**Reply by keunes** [reply]
@comradekingu Would you have an useful insights here? I'm not sure what the file should look like; what Weblate 'expects'. Should the English text be both the `msgid` _and_ `msgstr`? (If so, I should create an issue with `mdpo`.)

**Reply by comradekingu** [reply]
@keunes
Depends, if it is used as a language in itself then yes.
If it is just used as a base language to translate into en_US and en_UK then no.(?)
I think the latter is preferable, and it is possible to auto-populate them either manually, or in Weblate.

The idea is to not end up with missing text for untranslated strings, and there are setting for that on the whole.

https://docs.weblate.org/en/latest/formats.html?highlight=msgid#gettext
https://docs.weblate.org/en/latest/devel/gettext.html?highlight=msgid#extracting-translatable-strings

**Comment by keunes** [comment, marked as answer]
Ok, I've been using POT files instead and everything seems to work as expected now.
